### PR TITLE
Add scalene

### PR DESCRIPTION
**Pull request by jonashaag**
Generated with grayskull.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by jonashaag** [comment]
@conda-forge-admin restart CI

**Comment by jonashaag** [comment]
OSX build fails because of unsupported `-march=core2`, how to remove? It's injected by some Conda/Conda-forge tool.

How to add OSX arm64?

**Comment by dopplershift** [comment]
I'm not sure about the build failure, but osx arm64 can be [added](https://conda-forge.org/docs/maintainer/knowledge_base.html?highlight=arm64#apple-silicon-builds) once this recipe is merged.

File: recipes/scalene/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+test:
+  imports:
+    - scalene
+  commands:
+    - scalene --help
+  requires:
+    - pip
+
+about:
+  home: https://github.com/emeryberger/scalene
```
Comment (fanninpm) [inline recipes/scalene/meta.yaml:44]:
```suggestion
  home: https://github.com/plasma-umass/scalene
```

File: recipes/scalene/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
+{% set name = "scalene" %}
+{% set version = "1.3.16" %}
```
Comment (fanninpm) [inline recipes/scalene/meta.yaml:2]: Can this be updated? Currently, the latest is `v1.5.9.1`.

Comment (jonashaag) [reply recipes/scalene/meta.yaml]: Maybe this will help fix the build...

File: recipes/scalene/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/scalene-{{ version }}.tar.gz
+  sha256: a3adfe397e1e379091e3405c692a178291ebe8c2d3721dbc02bfec5ecc347ef7
+
+build:
+  number: 0
+  skip: true   # [py<37]
```
Comment (fanninpm) [inline recipes/scalene/meta.yaml:15]:
```suggestion
  skip: true   # [py<38]
```

Comment (jonashaag) [reply recipes/scalene/meta.yaml:15]: https://github.com/plasma-umass/scalene/blob/master/setup.py#L169?

Comment (fanninpm) [reply recipes/scalene/meta.yaml:15]: [Support for Python 3.7 has been removed.](https://github.com/plasma-umass/scalene/commit/9b7cb2dfe4bd3f2da6f372c1bf5b3bd95ea4b333)

File: recipes/scalene/meta.yaml (hunk 2 above)
Comment (fanninpm) [reply recipes/scalene/meta.yaml]: My mistake. [The PyPI version](https://pypi.org/project/scalene/1.5.9/) does not match [the GitHub version](https://github.com/plasma-umass/scalene/releases/tag/v1.5.9.1).

File: recipes/scalene/meta.yaml (hunk 3 above)
Comment (jonashaag) [reply recipes/scalene/meta.yaml:15]: Oops

Comment (jonashaag) [reply recipes/scalene/meta.yaml]: My brain had a bug evaluating `<` and `>=`

**Comment by jonashaag** [comment]
@fanninpm

**Comment by fanninpm** [comment]
Have you replicated the problem with [`python build-locally.py`](https://conda-forge.org/docs/maintainer/adding_pkgs.html#running-tests-locally-for-staged-recipes)?

File: recipes/scalene/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/scalene-{{ version }}.tar.gz
+  sha256: a3adfe397e1e379091e3405c692a178291ebe8c2d3721dbc02bfec5ecc347ef7
+
+build:
+  number: 0
+  skip: true   # [py<38]
+  entry_points:
+    - scalene = scalene.__main__:main
+  script: make bdist #{{ PYTHON }} -m pip install . -vv
```
Comment (fanninpm) [inline recipes/scalene/meta.yaml:18]: I'm going to go out on a limb and say that we don't need to `make` a `bdist`.

```suggestion
  script: {{ PYTHON }} -m pip install . -vv
```

**Comment by jonashaag** [comment]
> Have you replicated the problem with [python build-locally.py](https://conda-forge.org/docs/maintainer/adding_pkgs.html#running-tests-locally-for-staged-recipes)?

No, I was hoping you know what the error is about. :)

The `make bdist` was just a desperate guess taken from the upstream build instructins.

**Pull request closed without merging**